src/window.ts

```diff
@@ -112,24 +114,66 @@ export type SMonsterEventNotLive = {
 
 export type SMonsterEvent = BetterUXWrapper<SMonsterEventNotLive | SMonsterEventLive>;
 
+export type SEventBooleanKeys =
+  | "egghunt"
+  | "halloween"
+  | "holidayseason"
+  | "lunarnewyear"
+  | "valentines";
+
+export type SPartialEventBoolean = Partial<Record<Extract<SEventBooleanKeys, EventKey>, boolean>>;
+
+// TODO: should we just use MonsterKey?
+export type SPartialEventMonsterEvent = Partial<
+  Record<Exclude<EventKey, SEventBooleanKeys | "abtesting" | "goobrawl">, SMonsterEvent>
+>;
+export type SPartialHolidaySeasonMonsters = Partial<Record<"grinch" | "snowman", SMonsterEvent>>;
+export type SPartialHalloweenMonsters = Partial<
+  Record<"mrpumpkin" | "mrgreen" | "slenderman", SMonsterEvent>
+>;
+
+export type SPartialABTesting = Partial<
+  Record<
+    "abtesting",
+    {
+      /** A date string of when the event will end */
+      end?: string;
+
+      /** A date string of when sign-ups will stop for the event */
+      signup_end: string;
+      /** How many characters are on team A. TODO: Check that this is actually what it means */
+      A: number;
+      /** How many characters are on team B. TODO: Check that this is actually what it means */
+      B: number;
+      /** The event ID. TODO: Do we need this? */
+      id: string;
+    }
+  >
+>;
+
+export type SPartialGooBrawlEventInfo = Partial<
+  Record<
+    "abtesting",
+    {
+      /** A date string of when the event will end */
+      end?: string;
+    }
+  >
+>;
+
 export type SEventsInfos = {
   schedule: {
     time_offset: number;
     dailies: Array<number>;
     nightlies: Array<number>;
     night: boolean;
   };
-} & Partial<Record<EventKey, SMonsterEvent>> & {
-    // Christmas
-    holidayseason?: boolean; // unsure if this will override the above general purpose EventKey
-  } & Partial<Record<"grinch" | "snowman", SMonsterEvent>> & {
-    valentines?: boolean;
-  } & {
-    // halloween
-    halloween?: boolean; // unsure if this will override the above general purpose EventKey
-  } & Partial<Record<"mrpumpkin" | "mrgreen", SMonsterEvent>> & {
-    crabxx?: SMonsterEvent;
-  };
+} & SPartialEventBoolean &
```

Comment:
I also think this is reaching a point where it should be split into a dedicated file.

And I'm starting to think a `parent` or `window` folder would not hurt since this file has a lot of things!

Reply:
I am totally on board with that!